### DISCUSSION TITLE
How to display clock?

### DISCUSSION THREAD

**Discussion by DavidIgor** [cedricp/ddt4all#1006] [Q&A]
I want to have clock on radio what should be written in red circle? ![image](https://github.com/cedricp/ddt4all/assets/122531259/532a82fe-0d8c-4f43-b0fc-1f6211200010)!

**Comment by KarelSvo** [comment]
You can't say that there are no currently read data.

**Reply by DavidIgor** [reply]
@KarelSvo In was able to change adaptive volume curve... but this one is weird for me ... so you saying it is not possible?

**Reply by KarelSvo** [reply]
elm_log.txt file

**Reply by DavidIgor** [reply]
you refer to yesterday's file? I have modified my elm to receive audio modules(installed switch) and successfully connected to the "audio" today!

**Reply by DavidIgor** [reply]
haven't got time for this... but here i am again! so that is read data "00" that what is written in radio!  Im still lost with bits ... so please what should I type to get Clock(time) displayed?  also how it can be "00" does it only means "antenna passive" ? 
thanks

**Reply by DavidIgor** [reply]
@AndyIreland @KarelSvo

**Reply by DavidIgor** [reply]
@KarelSvo here: 
[elm_log.txt](https://github.com/cedricp/ddt4all/files/14967507/elm_log.txt)

**Comment by KarelSvo** [comment]
To answer your initial question, the ECU answer to 226C1C is necessary.

**Comment by DavidIgor** [comment]
[elm_log.txt](https://github.com/cedricp/ddt4all/files/13941705/elm_log.txt)
here ...

**Comment by KarelSvo** [comment]
Is not included. This ECU screen was not read.

**Comment by DavidIgor** [comment]
im lost ....  I have connected to audio module with ddt4all and also with pyren  and have changed that volume curve from "speed3" to "speed5" (even verified it after). 
Im too dumb to come up with those bits for displaying clock on radio screen - so I was in that someone here  will just tell me to write "03" or whatever ... 
Im putting it on hold for now .... maybe tomorrow. 
Yeah tomorrow ill read radio ecu screen again .

**Reply by AndyIreland** [reply]
maybe you are in the wrong ecu, maybe you need to be in the navigation ecu's ... mind you your car has not got a MFD display has it (multifunctional touch screen) ?

**Reply by KarelSvo** [reply]
>so I was in that someone here will just tell me to write "03" or whatever ...

A byte consists of 8 bits. Only 1 bit is changed for the clock.
So you first have to know whether the other 7 bits original are 0 or 1.
Otherwise you can't calculate which byte needs to be sent.
Or you are sending an incorrect configuration.

elm_ddt file

**Reply by DavidIgor** [reply]
@AndyIreland this one ![image](https://github.com/cedricp/ddt4all/assets/122531259/845a4803-a6b9-4a32-860c-2cda220064e9)

**Reply by AndyIreland** [reply]
yep thats not a touch screen multi function display (MFD) so just 'radio' 

What you could do (at your own risk) and this could be dangerous but without choosing any car , go into 'Audio' ECU folder there in DDT4ALL - no at this point do not write anything/any parameters and do not click on 'einstein' to make changes but just read and see what radio's respond. Obviously if when you read everything is in RED and says something like 'NO DATA' then that is not the right ECU but if you come across a radio ECU that does look like it has read the parameters correctly and it has a setting for 'CLOCK/TIME" and its not in red and says 'diasbled' or 'without' then and I say this again if you are feeling brave (or stupid) you could try setting it to 'enabled' then Hard reset the radio (do not use 'soft reset!) and see if that gets the clock displaying - as I say DO THIS AT YOUR OWN RISK and dont forget if you 'brick' / corrupt / break your radio ... you did not hear this from me! :D 

![image](https://github.com/cedricp/ddt4all/assets/49172217/444628da-22ed-4693-b7dc-9deb94cf19b8)

**Reply by DavidIgor** [reply]
@KarelSvo so for sake of learning (me understanding) parameter above was on "03" 
how can I understand it? 
![image](https://github.com/cedricp/ddt4all/assets/122531259/626a8e5c-2944-434e-97da-5d4eab93991e)

**Reply by DavidIgor** [reply]
@AndyIreland I have looked into this earlier today... but haven't found radio ecu with clock option. maybe this type of radio can not display clock!? 
will see tomorrow! 
It would be nice to have clock there and dashboard display can be used for something more useful than just displaying clock (as I have Start and stop system).

**Reply by AndyIreland** [reply]
I would be very surprised if your radio there has not got RDS radio text system - I zoomed in and on preset button 5 underneath it says 'INFO' have you tried pressing (or long pressing) this button to see if on each press it gives you a choice of displaying what song and artist is playing on the radio, and on another press - the radio station name, and on another press the time / time & date showing? - have you tried that. if it does then you dont have to bother adjusting anything in DDT4ALL to get the time to display on the screen (this is presuming the radio station is broadcasting the RDS signal)

**Reply by DavidIgor** [reply]
@AndyIreland that is something for tomorrow morning... will report back

**Reply by AndyIreland** [reply]
If ECU Shows 'NO DATA' in red box on ECU's - either wrong ECU selected/radio ECU , or that parameter cannot be read, or you need to increase the CAN Timeout to 600ms at the bottom of DDT4ALL page and refresh/try to read again

**Reply by DavidIgor** [reply]
@AndyIreland screenshot was after hitting scann can2 ... it found only that "audio" ecu or module. 
so maybe timeout problem. 
will test tomorrow!

**Reply by KarelSvo** [reply]
>so for sake of learning (me understanding) parameter above was on "03"
how can I understand it?

Exact answers can i give with the original data in the photo or the log file.
The radio file is correct, it is an R010. This radio also has additional functions if you press and longer hold the menu button. I recommend only using PyRen, it has a backup function.

**Reply by DavidIgor** [reply]
@AndyIreland that info button (5) does nothing... it only shows name of interpreter when media played! 
in radio menu I can choose to see rds info but only song name !

**Reply by AndyIreland** [reply]
ah right, if info only shows song name then either the radio is not RDS EON enabled or the station itself is not broadcasting the time and date on their RDS transmitting system, I have come across that myself before where they were not transmitting the time through their RDS signal but just station and song info.

**Reply by DavidIgor** [reply]
@AndyIreland yeah nah ! it is the radio in my car! 
I hoped to somehow display clock there , but it is not possible.

**Reply by DavidIgor** [reply]
All right... so that curve that I was in it was speed curve volume thing is actually a volume sensitivity. nothing related to speed or the car. 

what a radio!

**Comment by KarelSvo** [comment]
> I hoped to somehow display clock there , but it is not possible.

How did you find out?
What did you write in the ECU?
My questions about the original condition remained unanswered.
